tests/test_bonfire.py

```diff
@@ -0,0 +1,228 @@
+    mocker.patch("bonfire.openshift.parse_restype", return_value="")
+    mocker.patch("bonfire.openshift.get_json", return_value="")
+    mocker.patch("bonfire.openshift.get_all_reservations", return_value="")
+    mocker.patch("bonfire.bonfire.reserve_namespace", return_value=ns)
+
+    runner = CliRunner()
+    result = runner.invoke(bonfire.namespace, ["reserve", "--duration", duration])
+
+    assert result.output.rstrip() == expected
+
+
+def test_ns_list_option(mocker):
+    all_namespaces = []
+
+    namespace_1 = Mock(reserved=False, status="ready", clowdapps="none")
+    namespace_1.name = "namespace-1"
+    namespace_1.requester = "user-2"
+    namespace_1.expires_in = "2h"
+
+    namespace_2 = Mock(reserved=True, status="ready", clowdapps="none")
+    namespace_2.name = "namespace-2"
+    namespace_2.requester = "user-1"
+    namespace_2.expires_in = "31m"
+
+    namespace_3 = Mock(reserved=False, status="ready", clowdapps="none")
+    namespace_3.name = "namespace-3"
+    namespace_3.requester = "user-3"
+    namespace_3.expires_in = "6h"
+
+    all_namespaces.append(namespace_1)
+    all_namespaces.append(namespace_2)
+    all_namespaces.append(namespace_3)
+
+    mocker.patch("bonfire.openshift.get_all_reservations", return_value="")
+    mocker.patch("bonfire.bonfire.has_ns_operator", return_value=True)
+    mocker.patch("bonfire.openshift.get_api_resources", return_value=[])
+    mocker.patch("bonfire.bonfire.get_namespaces", return_value=all_namespaces)
+
+    runner = CliRunner()
+    result = runner.invoke(bonfire.namespace, ["list"])
+
+    assert "namespace-1" in result.output
+    assert "namespace-2" in result.output
+    assert "namespace-3" in result.output
+    assert "user-1" in result.output
+    assert "user-2" in result.output
+    assert "user-3" in result.output
+    assert "31m" in result.output
+    assert "2h" in result.output
+    assert "6h" in result.output
+
+
+def test_ns_list_options_available(mocker):
+    all_namespaces = []
+    all_reservations = []
+
+    namespace_1 = Mock(reserved=False, status="ready", clowdapps="none")
+    namespace_1.name = "namespace-1"
+    namespace_1.requester = None
+    namespace_1.expires_in = None
+
+    namespace_2 = Mock(reserved=True, status="ready", clowdapps="none")
+    namespace_2.name = "namespace-2"
+    namespace_2.requester = "user-1"
+    namespace_2.expires_in = "31m"
+
+    namespace_3 = Mock(reserved=True, status="ready", clowdapps="none")
+    namespace_3.name = "namespace-3"
+    namespace_3.requester = None
+    namespace_3.expires_in = None
+
+    namespace_4 = Mock(reserved=True, status="ready", clowdapps="none")
+    namespace_4.name = "namespace-4"
+    namespace_4.requester = "user-2"
+    namespace_4.expires_in = "1h"
+
+    all_namespaces.append(namespace_1)
+    all_namespaces.append(namespace_2)
+    all_namespaces.append(namespace_3)
+    all_namespaces.append(namespace_4)
+
+    all_reservations.append(namespace_1)
```

Comment:
Similar comment to what I have about `mine=True` filter below as well.

I am not sure if this is truly asserting the behavior to check that the 'available=True' filtering is working in the get_namespaces function.

I think the more ideal thing to do would be to mock these function calls out:
https://github.com/RedHatInsights/bonfire/blob/master/bonfire/namespaces.py#L204-L206

Mock get_all_namespaces to return the 4 fake namespaces you're creating up top. Two of those fake namespaces should be set up such that their "available" function returns True: https://github.com/RedHatInsights/bonfire/blob/master/bonfire/namespaces.py#L120

And in that way, we'd be ensuring that get_namespaces(available=True) is only returning namespaces which are ready + not reserved